crates/serialize_hierarchy/src/lib.rs

```diff
@@ -262,6 +262,240 @@ impl<T> SerializeHierarchy for Vec<T> {
+    fn serialize_path<S>(&self, path: &str) -> Result<S::Serialized, Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let index = ["x", "y", "z"].into_iter().position(|name| name == path);
+        match index {
+            Some(index) => S::serialize(&self[index]).map_err(Error::SerializationFailed),
+            _ => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn deserialize_path<S>(
+        &mut self,
+        path: &str,
+        _data: S::Serialized,
+    ) -> Result<(), Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let index = ["x", "y", "z"].into_iter().position(|name| name == path);
+        match index {
+            Some(index) => {
+                let deserialized = S::deserialize(_data).map_err(Error::DeserializationFailed)?;
+                self[index] = deserialized;
+                Ok(())
+            }
+            None => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn exists(path: &str) -> bool {
+        matches!(path, "x" | "y" | "z")
+    }
+
+    fn get_fields() -> BTreeSet<String> {
+        ["x", "y", "z"].into_iter().map(String::from).collect()
+    }
+}
+
+impl<T: Serialize + DeserializeOwned> SerializeHierarchy for Vector4<T> {
+    fn serialize_path<S>(&self, path: &str) -> Result<S::Serialized, Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let index = ["x", "y", "z", "w"]
+            .into_iter()
+            .position(|name| name == path);
+        match index {
+            Some(index) => S::serialize(&self[index]).map_err(Error::SerializationFailed),
+            _ => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn deserialize_path<S>(
+        &mut self,
+        path: &str,
+        _data: S::Serialized,
+    ) -> Result<(), Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let index = ["x", "y", "z", "w"]
+            .into_iter()
+            .position(|name| name == path);
+        match index {
+            Some(index) => {
+                let deserialized = S::deserialize(_data).map_err(Error::DeserializationFailed)?;
+                self[index] = deserialized;
+                Ok(())
+            }
+            None => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn exists(path: &str) -> bool {
+        matches!(path, "x" | "y" | "z" | "w")
+    }
+
+    fn get_fields() -> BTreeSet<String> {
+        ["x", "y", "z", "w"].into_iter().map(String::from).collect()
+    }
+}
+
+impl<T: Serialize + DeserializeOwned + Clone + Scalar> SerializeHierarchy for Point2<T> {
```

Comment:
I think we can reduce some code duplication for `Point` since points contain [`coords: Vector`](https://docs.rs/nalgebra/latest/nalgebra/geometry/struct.OPoint.html#structfield.coords). It should be possible to delegate the trait method calls to the `coords` field instead of manually implementing it again, right?